rpcs3/Emu/Cell/Modules/sceNp.cpp

```diff
@@ -2416,24 +2574,28 @@ error_code sceNpManagerGetAvatarUrl(vm::ptr<SceNpAvatarUrl> avatarUrl)
 		return SCE_NP_ERROR_INVALID_ARGUMENT;
 	}
 
-	if (g_psn_connection_status == SCE_NP_MANAGER_STATUS_OFFLINE)
+	if (nph->get_psn_status() == SCE_NP_MANAGER_STATUS_OFFLINE)
 	{
 		return not_an_error(SCE_NP_ERROR_OFFLINE);
 	}
 
-	if (g_psn_connection_status != SCE_NP_MANAGER_STATUS_ONLINE)
+	if (nph->get_psn_status() != SCE_NP_MANAGER_STATUS_ONLINE)
 	{
 		return SCE_NP_ERROR_INVALID_STATE;
 	}
 
+	memcpy(avatarUrl.get_ptr(), &nph->get_avatar_url(), sizeof(SceNpAvatarUrl));
```

Comment:
Nitpick, but sizeof(nph->get_avatar_url()) may work. Or something similar, to tie the size to this function's return type.

Reply:
again, vm::ptr<>::size() (avatarUrl.size())